From ece235682c7582f9a6e1a331f682eaa2609aa85b Mon Sep 17 00:00:00 2001
From: kelson8 <kelson@kelsoncraft.net>
Date: Fri, 11 Sep 2026 11:05:25 -0400
Subject: [PATCH 126/161] refactor: Label some functions

* I labeled most functions in Pickups.cpp as to what I think they are doing.
* Label somethings in Pathfind.cpp.

* Label some functions and variables in Pathfind.cpp and World.cpp.

* Update changelog.
---
 Changelog.md             |  14 ++-
 src/control/PathFind.cpp |  42 +++++++
 src/control/Pickups.cpp  | 229 +++++++++++++++++++++++++++++++++++++++
 src/core/World.cpp       |   7 ++
 4 files changed, 291 insertions(+), 1 deletion(-)

diff --git a/Changelog.md b/Changelog.md
index e7c31d8a..3ca581a4 100644
--- a/Changelog.md
+++ b/Changelog.md
@@ -597,6 +597,8 @@ feat: Add everyone ignore and police ignore toggles
 
 ### 1.2.13-1a
 
+**Commit 1**
+
 feat: Add infinite ammo global for lua
 
 * I added an infinite ammo global into lua to easily be toggled on or off.
@@ -605,4 +607,14 @@ feat: Add infinite ammo global for lua
 
 * Move freeroam text into defines.cpp/.h, and add version to freeroam text display.
 
-* Update changelog, change version to 1.2.13-1a.
\ No newline at end of file
+* Update changelog, change version to 1.2.13-1a.
+
+**Commit 2**
+
+refactor: Label some functions
+
+* I labeled most functions in Pickups.cpp as to what I think they are doing.
+
+* Label some functions and variables in Pathfind.cpp and World.cpp.
+
+* Update changelog.
\ No newline at end of file
diff --git a/src/control/PathFind.cpp b/src/control/PathFind.cpp
index 80d40b45..adc4ac60 100644
--- a/src/control/PathFind.cpp
+++ b/src/control/PathFind.cpp
@@ -1121,6 +1121,20 @@ CPathFind::SwitchOffNodeAndNeighbours(int32 nodeId, bool disable)
 		}
 }
 
+/**
+ * Turn off the vehicle roads in an area.
+ * 
+ * I may eventually rename this to 'ToggleVehicleRoadsInArea' with a preprocessor.
+ * This takes a list of CVector2 values, 2 for an x, y, and z value.
+ * 
+ * @param x1
+ * @param x2
+ * @param y1
+ * @param y2
+ * @param z1
+ * @param z2
+ * @param disable If the roads should be disabled, if this is true the roads in that area are disabled for peds.
+ */
 void
 CPathFind::SwitchRoadsOffInArea(float x1, float x2, float y1, float y2, float z1, float z2, bool disable)
 {
@@ -1136,6 +1150,20 @@ CPathFind::SwitchRoadsOffInArea(float x1, float x2, float y1, float y2, float z1
 	}
 }
 
+/**
+ * Turn off the ped roads in an area.
+ * 
+ * I may eventually rename this to 'TogglePedRoadsInArea' with a preprocessor.
+ * This takes a list of CVector2 values, 2 for an x, y, and z value.
+ * 
+ * @param x1
+ * @param x2
+ * @param y1
+ * @param y2
+ * @param z1
+ * @param z2
+ * @param disable If the roads should be disabled, if this is true the roads in that area are disabled for peds.
+ */
 void
 CPathFind::SwitchPedRoadsOffInArea(float x1, float x2, float y1, float y2, float z1, float z2, bool disable)
 {
@@ -1245,6 +1273,20 @@ CPathFind::PedMarkRoadsBetweenLevelsInArea(float x1, float x2, float y1, float y
 	}
 }
 
+/**
+ * Find the node closest to the coordinates
+ * TODO Possibly use this for my vehicle spawner since it can get the vehicle stuck in a wall sometimes.
+ * TODO Figure out what some of these values do.
+ * 
+ * @param coors The coordinates to find a node near.
+ * @param type Seems to be either 'PATH_CAR' or 'PATH_PED'
+ * @param distLimit
+ * @param ignoreDisabled
+ * @param ignoreBetweenLevels
+ * @param ignoreBetweenLevels
+ * @param ignoreSelected 
+ * @param bWaterPath If the path is in water?
+ */
 int32
 CPathFind::FindNodeClosestToCoors(CVector coors, uint8 type, float distLimit, bool ignoreDisabled, bool ignoreBetweenLevels, bool ignoreSelected, bool bWaterPath)
 {
diff --git a/src/control/Pickups.cpp b/src/control/Pickups.cpp
index f6b1a9b9..34e5a6ef 100644
--- a/src/control/Pickups.cpp
+++ b/src/control/Pickups.cpp
@@ -58,12 +58,15 @@ uint32 CPickups::StaticCamStartTime;
 
 tPickupMessage CPickups::aMessages[NUMPICKUPMESSAGES];
 
+// The ammo for the pickup weapons, used with 'PICKUP_IN_SHOP' and the 'PICKUP_ONCE' pickup types.
 uint16 AmmoForWeapon[WEAPONTYPE_TOTALWEAPONS + 1] = { 0,  1,  1,  1,   1,   1,   1,   1,   1,   1,  1,  1, 8,   8,   8,    8, 8,   68, 24,
 													 32, 28, 20, 200, 120, 120, 120, 120, 120, 40, 28, 8, 300, 200, 1000, 1, 400, 36, 0 };
 
+// The ammo for the pickup weapons on the street, used with the 'PICKUP_ON_STREET' pickup types..
 uint16 AmmoForWeapon_OnStreet[WEAPONTYPE_TOTALWEAPONS + 1] = { 0,  1,  1,  1,   1,  1,  1,  1,  1,  1,  1,  1, 4,   4,   4,   4, 4,   34, 12,
 															  16, 14, 10, 100, 60, 60, 60, 60, 60, 20, 14, 4, 150, 100, 500, 1, 400, 36, 0 };
 
+// The cost of the pickup weapons, mostly used for 'PICKUP_IN_SHOP'
 uint16 CostOfWeapon[WEAPONTYPE_TOTALWEAPONS + 3] = { 0,     10,    10,   10,   10,   10,    10,   10,    10,  10,  10,  10,   1000, 1000,
 													1000,  500,   8000, 250,  400,  1200,  1250, 1250,  800, 800, 650, 1200, 5000, 400,
 													10000, 10000, 8000, 8000, 8000, 10000, 1000, 11000, 500, 20,  10,  0 };
@@ -120,6 +123,10 @@ struct
 };
 
 
+/**
+ * Seems to only run on 'PICKUP_PROPERTY_FORSALE'.
+ * TODO Try to figure out what this is doing.
+ */
 void
 ModifyStringLabelForControlSetting(char *str)
 {
@@ -146,6 +153,11 @@ ModifyStringLabelForControlSetting(char *str)
 	}
 }
 
+/**
+ * Extract the ammo for the weapon pickup.
+ * 
+ * @param player
+ */
 void
 CPickup::ExtractAmmoFromPickup(CPlayerPed *player)
 {
@@ -168,6 +180,9 @@ removeAmmo:
 	m_bWasAmmoCollected = true;
 }
 
+/**
+ * Remove the pickup.
+ */
 void
 CPickup::Remove()
 {
@@ -277,6 +292,9 @@ CPickup::GiveUsAPickUpObject(CObject **ppObject, CObject **ppExtraObject, int32
 	return object;
 }
 
+/**
+ * Check if the pickup can be picked up.
+ */
 bool
 CPickup::CanBePickedUp(CPlayerPed *player, int playerId)
 {
@@ -290,6 +308,13 @@ CPickup::CanBePickedUp(CPlayerPed *player, int playerId)
 	return !cannotBePickedUp;
 }
 
+/**
+ * Update the pickups
+ * 
+ * @param player
+ * @param vehicle
+ * @param playerId
+ */
 bool
 CPickup::Update(CPlayerPed *player, CVehicle *vehicle, int playerId)
 {
@@ -393,12 +418,15 @@ CPickup::Update(CPlayerPed *player, CVehicle *vehicle, int playerId)
 								{
 								case 0:
 								case 1:
+									// Press the ~h~~k~~PED_ANSWER_PHONE~ ~w~button to pick up this weapon. It will replace any weapon you have of the same type.
 									CHud::SetHelpMessage(TheText.Get("PU_CF1"), false);
 									break;
 								case 2:
+									// Press the ~h~~k~~PED_ANSWER_PHONE~ ~w~button to replace current weapon in this slot.
 									CHud::SetHelpMessage(TheText.Get("PU_CF3"), false);
 									break;
 								case 3:
+									// Press the ~h~~k~~PED_ANSWER_PHONE~ ~w~button to replace current weapon in this slot.
 									CHud::SetHelpMessage(TheText.Get("PU_CF4"), false);
 									break;
 								default:
@@ -428,6 +456,7 @@ CPickup::Update(CPlayerPed *player, CVehicle *vehicle, int playerId)
 			{
 			case PICKUP_IN_SHOP:
 				if (CWorld::Players[playerId].m_nMoney < CostOfWeapon[weaponType])
+					// You don't have enough cash.
 					CGarages::TriggerMessage("PU_MONY", -1, 6000, -1);
 				else {
 					CWorld::Players[playerId].m_nMoney -= CostOfWeapon[weaponType];
@@ -501,9 +530,11 @@ CPickup::Update(CPlayerPed *player, CVehicle *vehicle, int playerId)
 
 				if (CWorld::Players[playerId].m_nCollectedPackages == CWorld::Players[playerId].m_nTotalPackages) {
 					printf("All collectables have been picked up\n");
+					// You got all of them. Here's a little something...
 					CGarages::TriggerMessage("CO_ALL", -1, 5000, -1);
 					CWorld::Players[CWorld::PlayerInFocus].m_nMoney += 100000;
 				} else
+					// Hidden Package ~1~ of ~1~
 					CGarages::TriggerMessage("CO_ONE", CWorld::Players[CWorld::PlayerInFocus].m_nCollectedPackages, 5000, CWorld::Players[CWorld::PlayerInFocus].m_nTotalPackages);
 
 				result = true;
@@ -541,6 +572,7 @@ CPickup::Update(CPlayerPed *player, CVehicle *vehicle, int playerId)
 				if (CollectPickupBuffer == 0)
 					break;
 				if (CTheScripts::IsPlayerOnAMission())
+					// You cannot buy property whilst on a mission
 					CHud::SetHelpMessage(TheText.Get("PROP_2"), true);
 				else {
 					if (CWorld::Players[CWorld::PlayerInFocus].m_nMoney >= m_nQuantity) {
@@ -550,6 +582,7 @@ CPickup::Update(CPlayerPed *player, CVehicle *vehicle, int playerId)
 						Remove();
 						break;
 					}
+					// You don't have enough cash for this property
 					CHud::SetHelpMessage(TheText.Get("PROP_1"), true);
 				}
 				break;
@@ -654,6 +687,11 @@ CPickup::Update(CPlayerPed *player, CVehicle *vehicle, int playerId)
 	return result;
 }
 
+/**
+ * Process a gun shot for the pickups
+ * 
+ * TODO Look into this, the CCollision::TestLineSphere function seems useful.
+ */
 void
 CPickup::ProcessGunShot(CVector *vec1, CVector *vec2)
 {
@@ -673,6 +711,9 @@ CPickup::ProcessGunShot(CVector *vec1, CVector *vec2)
 	}
 }
 
+/**
+ * This looks like it gets rid of all the objects spawned in.
+ */
 void
 CPickup::GetRidOfObjects()
 {
@@ -688,6 +729,9 @@ CPickup::GetRidOfObjects()
 	}
 }
 
+/**
+ * Init for the pickups.
+ */
 void
 CPickups::Init(void)
 {
@@ -705,6 +749,16 @@ CPickups::Init(void)
 	CollectedPickUpIndex = 0;
 }
 
+/**
+ * Check if pickups are in the specified range.
+ * TODO Test this later.
+ * 
+ * Seems to mostly be used for the 'IS_ANY_PICKUP_AT_COORDS' command.
+ * https://library.sannybuilder.com/#/vc/script/extensions/default/048C
+ * 
+ * @param pos The position to check.
+ * @param range The range of the area to check if pickups are in.
+ */
 bool
 CPickups::TestForPickupsInBubble(CVector pos, float range)
 {
@@ -727,6 +781,11 @@ CPickups::TryToMerge_WeaponType(CVector pos, eWeaponType weapon, uint8 type, uin
 	return false;
 }
 
+/**
+ * Check if a pickup has been picked up.
+ * 
+ * @param pickupId The item pickup to check.
+ */
 bool
 CPickups::IsPickUpPickedUp(int32 pickupId)
 {
@@ -739,6 +798,9 @@ CPickups::IsPickUpPickedUp(int32 pickupId)
 	return false;
 }
 
+/**
+ * Pass the time for the pickups.
+ */
 void
 CPickups::PassTime(uint32 time)
 {
@@ -752,6 +814,9 @@ CPickups::PassTime(uint32 time)
 	}
 }
 
+/**
+ * Get the pickup index.
+ */
 int32
 CPickups::GetActualPickupIndex(int32 index)
 {
@@ -762,6 +827,12 @@ CPickups::GetActualPickupIndex(int32 index)
 	return (uint16)index;
 }
 
+/**
+ * Give the player the pickup effects, and play the sound.
+ * 
+ * @param modelIndex Seems to be a value such as this 'MI_PICKUP_BODYARMOUR' in ModelIndices.h.
+ * @param playerIndex The player to give the pickup effect to.
+ */
 bool
 CPickups::GivePlayerGoodiesWithPickUpMI(int16 modelIndex, int playerIndex)
 {
@@ -770,31 +841,44 @@ CPickups::GivePlayerGoodiesWithPickUpMI(int16 modelIndex, int playerIndex)
 	if (playerIndex <= 0) player = CWorld::Players[CWorld::PlayerInFocus].m_pPed;
 	else player = CWorld::Players[playerIndex].m_pPed;
 
+		// Adrenaline
 	if (modelIndex == MI_PICKUP_ADRENALINE) {
 		player->m_bAdrenalineActive = true;
 		player->m_nAdrenalineTime = CTimer::GetTimeInMilliseconds() + 20000;
 		player->m_fCurrentStamina = player->m_fMaxStamina;
 		DMAudio.PlayFrontEndSound(SOUND_PICKUP_ADRENALINE, 0);
 		return true;
+
+		// Body Armor
 	} else if (modelIndex == MI_PICKUP_BODYARMOUR) {
 		player->m_fArmour = CWorld::Players[playerIndex].m_nMaxArmour;
 		DMAudio.PlayFrontEndSound(SOUND_PICKUP_ARMOUR, 0);
 		return true;
+
+		// Info
 	} else if (modelIndex == MI_PICKUP_INFO) {
 		DMAudio.PlayFrontEndSound(SOUND_PICKUP_BONUS, 0);
 		return true;
+
+		// Health
 	} else if (modelIndex == MI_PICKUP_HEALTH) {
 		player->m_fHealth = CWorld::Players[playerIndex].m_nMaxHealth;
 		DMAudio.PlayFrontEndSound(SOUND_PICKUP_HEALTH, 0);
 		return true;
+
+		// Bonus
 	} else if (modelIndex == MI_PICKUP_BONUS) {
 		DMAudio.PlayFrontEndSound(SOUND_PICKUP_BONUS, 0);
 		return true;
+
+		// Police bribe
 	} else if (modelIndex == MI_PICKUP_BRIBE) {
 		int32 level = Max(FindPlayerPed()->m_pWanted->GetWantedLevel() - 1, 0);
 		player->SetWantedLevel(level);
 		DMAudio.PlayFrontEndSound(SOUND_PICKUP_BONUS, 0);
 		return true;
+
+		// Rampage
 	} else if (modelIndex == MI_PICKUP_KILLFRENZY) {
 		DMAudio.PlayFrontEndSound(SOUND_PICKUP_BONUS, 0);
 		return true;
@@ -802,6 +886,11 @@ CPickups::GivePlayerGoodiesWithPickUpMI(int16 modelIndex, int playerIndex)
 	return false;
 }
 
+/**
+ * Remove a pickup from the world
+ * 
+ * @param pickupIndex The item pickup to remove.
+ */
 void
 CPickups::RemovePickUp(int32 pickupIndex)
 {
@@ -813,6 +902,7 @@ CPickups::RemovePickUp(int32 pickupIndex)
 		delete aPickUps[index].m_pObject;
 		aPickUps[index].m_pObject = nil;
 	}
+
 	if (aPickUps[index].m_pExtraObject) {
 		CWorld::Remove(aPickUps[index].m_pExtraObject);
 		delete aPickUps[index].m_pExtraObject;
@@ -822,6 +912,17 @@ CPickups::RemovePickUp(int32 pickupIndex)
 	aPickUps[index].m_bRemoved = true;
 }
 
+/**
+ * Generate a new pickup at the specified location
+ * 
+ * @param pos The position to spawn the pickup. 
+ * @param modelIndex Seems to be a value such as this 'MI_PICKUP_BODYARMOUR' in ModelIndices.h.
+ * @param type The pickup type to spawn from the ePickupType enum.
+ * @param quantity Possibly the amount of pickups to spawn.
+ * @param rate Possibly the rate to spawn the pickups.
+ * @param highPriority This sets 'm_bWasAmmoCollected' on the pickup slot.
+ * @param pText The text for the pickups, not sure what this is for.
+ */
 int32
 CPickups::GenerateNewOne(CVector pos, uint32 modelIndex, uint8 type, uint32 quantity, uint32 rate, bool highPriority, char* pText)
 {
@@ -870,20 +971,27 @@ CPickups::GenerateNewOne(CVector pos, uint32 modelIndex, uint8 type, uint32 quan
 	aPickUps[slot].m_nTimer = CTimer::GetTimeInMilliseconds();
 	aPickUps[slot].m_bWasAmmoCollected = highPriority;
 	aPickUps[slot].m_bWasControlMessageShown = false;
+
 	if (type == PICKUP_ONCE_TIMEOUT)
 		aPickUps[slot].m_nTimer = CTimer::GetTimeInMilliseconds() + 20000;
+
 	else if (type == PICKUP_ONCE_TIMEOUT_SLOW)
 		aPickUps[slot].m_nTimer = CTimer::GetTimeInMilliseconds() + 120000;
+
 	else if (type == PICKUP_MONEY)
 		aPickUps[slot].m_nTimer = CTimer::GetTimeInMilliseconds() + 30000;
+
 	else if (type == PICKUP_MINE_INACTIVE || type == PICKUP_MINE_ARMED) {
 		aPickUps[slot].m_eType = PICKUP_MINE_INACTIVE;
 		aPickUps[slot].m_nTimer = CTimer::GetTimeInMilliseconds() + 1500;
+
 	} else if (type == PICKUP_NAUTICAL_MINE_INACTIVE || type == PICKUP_NAUTICAL_MINE_ARMED) {
 		aPickUps[slot].m_eType = PICKUP_NAUTICAL_MINE_INACTIVE;
 		aPickUps[slot].m_nTimer = CTimer::GetTimeInMilliseconds() + 1500;
+
 	}
 	aPickUps[slot].m_eModelIndex = modelIndex;
+
 	if (pText)
 		strncpy(aPickUps[slot].m_sTextKey, pText, 8);
 	else
@@ -891,19 +999,37 @@ CPickups::GenerateNewOne(CVector pos, uint32 modelIndex, uint8 type, uint32 quan
 
 	aPickUps[slot].m_vecPos = pos;
 	aPickUps[slot].m_pObject = aPickUps[slot].GiveUsAPickUpObject(&aPickUps[slot].m_pObject, &aPickUps[slot].m_pExtraObject, -1, -1);
+
 	if (aPickUps[slot].m_pObject)
 		CWorld::Add(aPickUps[slot].m_pObject);
+
 	if (aPickUps[slot].m_pExtraObject)
 		CWorld::Add(aPickUps[slot].m_pExtraObject);
+
 	return GetNewUniquePickupIndex(slot);
 }
 
+/**
+ * Generate a new pickup with a specific weapon type
+ * 
+ * @param pos The position to generate the pickup.
+ * @param weaponType The weapon type to spawn from the eWeaponType enum.
+ * @param type The type, from the 'ePickupType' enum.
+ * @param quantity Possibly the amount of ammo for the weapon pickup.
+ */
 int32
 CPickups::GenerateNewOne_WeaponType(CVector pos, eWeaponType weaponType, uint8 type, uint32 quantity)
 {
 	return GenerateNewOne(pos, ModelForWeapon(weaponType), type, quantity);
 }
 
+/**
+ * Get a new unique pickup index.
+ * TODO Figure out what this does.
+ * This seems to only be used by 'CPickups::GenerateNewOne'
+ * 
+ * @param slot
+ */
 int32
 CPickups::GetNewUniquePickupIndex(int32 slot)
 {
@@ -914,12 +1040,22 @@ CPickups::GetNewUniquePickupIndex(int32 slot)
 	return slot | (aPickUps[slot].m_nIndex << 16);
 }
 
+/**
+ * Get the model for a weapon
+ * 
+ * @param weaponType The weapon type to get the model from the eWeaponType enum.
+ */
 int32
 CPickups::ModelForWeapon(eWeaponType weaponType)
 {
 	return CWeaponInfo::GetWeaponInfo(weaponType)->m_nModelId;
 }
 
+/**
+ * Get the weapon type for a pickup.
+ * 
+ * @param model The model type for the pickup.
+ */
 eWeaponType
 CPickups::WeaponForModel(int32 model)
 {
@@ -930,6 +1066,13 @@ CPickups::WeaponForModel(int32 model)
 	return ((CWeaponModelInfo*)CModelInfo::GetModelInfo(model))->GetWeaponInfo();
 }
 
+/**
+ * Add the pickup to the collected index.
+ * 
+ * Seems to only be used in the CPickups::Update function.
+ * 
+ * @param index
+ */
 void
 CPickups::AddToCollectedPickupsArray(int32 index)
 {
@@ -938,6 +1081,9 @@ CPickups::AddToCollectedPickupsArray(int32 index)
 		CollectedPickUpIndex = 0;
 }
 
+/**
+ * Update and process the pickups logic.
+ */
 void
 CPickups::Update()
 {
@@ -1007,6 +1153,11 @@ CPickups::FindPickUpForThisObject(CEntity *object)
 	return &aPickUps[0];
 }
 
+/**
+ * Run the effects when the pickup is picked up.
+ * 
+ * @param entity
+ */
 void
 CPickups::DoPickUpEffects(CEntity *entity)
 {
@@ -1223,6 +1374,11 @@ CPickups::DoPickUpEffects(CEntity *entity)
 	}
 }
 
+/**
+ * Run the effects for the mines.
+ * 
+ * @param entity
+ */
 void
 CPickups::DoMineEffects(CEntity *entity)
 {
@@ -1242,6 +1398,11 @@ CPickups::DoMineEffects(CEntity *entity)
 	entity->GetMatrix().SetRotateZOnly((float)(CTimer::GetTimeInMilliseconds() & 0x3FF) * DEGTORAD(360.0f / 0x400));
 }
 
+/**
+ * Run the effects for money.
+ * 
+ * @param entity
+ */
 void
 CPickups::DoMoneyEffects(CEntity *entity)
 {
@@ -1261,6 +1422,11 @@ CPickups::DoMoneyEffects(CEntity *entity)
 	entity->GetMatrix().SetRotateZOnly((float)(CTimer::GetTimeInMilliseconds() & 0x7FF) * DEGTORAD(360.0f / 0x800));
 }
 
+/**
+ * Run the effects for the collectables.
+ * 
+ * @param entity
+ */
 void
 CPickups::DoCollectableEffects(CEntity *entity)
 {
@@ -1280,6 +1446,11 @@ CPickups::DoCollectableEffects(CEntity *entity)
 	entity->GetMatrix().SetRotateZOnly((float)(CTimer::GetTimeInMilliseconds() & 0xFFF) * DEGTORAD(360.0f / 0x1000));
 }
 
+/**
+ * Render the pickup text for some pickups, seems to mostly be in use for clothes pickups.
+ * TODO Make this use an enum.
+ * TODO Try to use this in my own pickups once I figure those out.
+ */
 void
 CPickups::RenderPickUpText()
 {
@@ -1298,6 +1469,7 @@ CPickups::RenderPickUpText()
 					strToPrint = nil;
 				} else {
 					if (aMessages[i].m_bOutOfStock)
+						// ~r~out of stock
 						strToPrint = TheText.Get("STOCK");
 					else {
 						sprintf(gString, "$%d", CostOfWeapon[aMessages[i].m_weaponType]);
@@ -1307,42 +1479,55 @@ CPickups::RenderPickUpText()
 				}
 				break;
 			case 1:
+				// Street
 				strToPrint = TheText.Get("OUTFT1");
 				break;
 			case 2:
+				// Soiree
 				strToPrint = TheText.Get("OUTFT2");
 				break;
 			case 3:
+				// Coveralls
 				strToPrint = TheText.Get("OUTFT3");
 				break;
 			case 4:
+				// Country Club
 				strToPrint = TheText.Get("OUTFT4");
 				break;
 			case 5:
+				// Havana
 				strToPrint = TheText.Get("OUTFT5");
 				break;
 			case 6:
+				// Cop
 				strToPrint = TheText.Get("OUTFT6");
 				break;
 			case 7:
+				// Bank Job
 				strToPrint = TheText.Get("OUTFT7");
 				break;
 			case 8:
+				// Casual
 				strToPrint = TheText.Get("OUTFT8");
 				break;
 			case 9:
+				// Mr Vercetti
 				strToPrint = TheText.Get("OUTFT9");
 				break;
 			case 10:
+				// Black Tracksuit
 				strToPrint = TheText.Get("OUTFT10");
 				break;
 			case 11:
+				// Red Tracksuit
 				strToPrint = TheText.Get("OUTFT11");
 				break;
 			case 12:
+				// Frankie
 				strToPrint = TheText.Get("OUTFT12");
 				break;
 			case 13:
+				// MC Tommy
 				strToPrint = TheText.Get("OUTFT13");
 				break;
 			default:
@@ -1379,6 +1564,12 @@ CPickups::RenderPickUpText()
 	NumMessages = 0;
 }
 
+/**
+ * Create a money pickup at the specified location.
+ * 
+ * @param pos The location to spawn a money pickup.
+ * @param money The amount of money to spawn.
+ */
 void
 CPickups::CreateSomeMoney(CVector pos, int money)
 {
@@ -1398,6 +1589,11 @@ CPickups::CreateSomeMoney(CVector pos, int money)
 	}
 }
 
+/**
+ * Remove all weapon pickups of the specified type.
+ * 
+ * @param weaponType The weapon from the eWeaponType to remove. 
+ */
 void
 CPickups::RemoveAllPickupsOfACertainWeaponGroupWithNoAmmo(eWeaponType weaponType)
 {
@@ -1420,6 +1616,12 @@ CPickups::RemoveAllPickupsOfACertainWeaponGroupWithNoAmmo(eWeaponType weaponType
 	}
 }
 
+/**
+ * Make mines detonate when hit by a gun shot.
+ * 
+ * @param vec1
+ * @param vec2
+ */
 void
 CPickups::DetonateMinesHitByGunShot(CVector *vec1, CVector *vec2)
 {
@@ -1429,6 +1631,14 @@ CPickups::DetonateMinesHitByGunShot(CVector *vec1, CVector *vec2)
 	}
 }
 
+/**
+ * Remove extra pickups, this is some type of cleanup function for them.
+ * 
+ * Seems to only be used in 'CWorld::ClearExcitingStuffFromArea'
+ * 
+ * @param center
+ * @param radius
+ */
 void
 CPickups::RemoveUnnecessaryPickups(const CVector& center, float radius)
 {
@@ -1443,6 +1653,11 @@ CPickups::RemoveUnnecessaryPickups(const CVector& center, float radius)
 	}
 }
 
+/**
+ * Load the pickups from the save.
+ * 
+ * TODO Look into this for my json save/load system.
+ */
 void
 CPickups::Load(uint8 *buf, uint32 size)
 {
@@ -1495,6 +1710,11 @@ INITSAVEBUF
 VALIDATESAVEBUF(size)
 }
 
+/**
+ * Save the pickups to the save.
+ * 
+ * TODO Look into this for my json save/load system.
+ */
 void
 CPickups::Save(uint8 *buf, uint32 *size)
 {
@@ -1645,6 +1865,9 @@ CPacManPickups::ResetPowerPillsCarriedByPlayer()
 {
 }
 
+/**
+ * Spawn money when the ped dies.
+ */
 void
 CPed::CreateDeadPedMoney(void)
 {
@@ -1665,6 +1888,9 @@ CPed::CreateDeadPedMoney(void)
 	m_nPedMoney = 0;
 }
 
+/**
+ * Spawn a weapon when the ped dies.
+ */
 void
 CPed::CreateDeadPedWeaponPickups(void)
 {
@@ -1690,6 +1916,9 @@ CPed::CreateDeadPedWeaponPickups(void)
 	ClearWeapons();
 }
 
+/**
+ * Create a pickup at the coordinates where the ped has died.
+ */
 void
 CPed::CreateDeadPedPickupCoors(float *x, float *y, float *z)
 {
diff --git a/src/core/World.cpp b/src/core/World.cpp
index 549c2cc0..1979f3e5 100644
--- a/src/core/World.cpp
+++ b/src/core/World.cpp
@@ -112,6 +112,13 @@ CWorld::ClearScanCodes(void)
 		}
 }
 
+/**
+ * Clear all peds, objects, and vehicles from the world in the specified area
+ * 
+ * @param pos The position to remove entities from.
+ * @param radius The radius to remove entities from.
+ * @param bRemoveProjectilesAndTidyUpShadows If this should remove all projectiles and shadows.
+ */
 void
 CWorld::ClearExcitingStuffFromArea(const CVector &pos, float radius, bool bRemoveProjectilesAndTidyUpShadows)
 {
-- 
2.43.0

